### DISCUSSION TITLE
improved SSAO

### DISCUSSION THREAD

**Discussion by GoldenThumbs** [nem0/LumixEngine#1342] [General]
I've been working on an improved SSAO method for lumix. It's a variation of HBAO, implemented in a compute shader.
You can see my progress on it [here](https://github.com/GoldenThumbs/LumixEngine/tree/gld_dev)

Screenshots:
![image](https://user-images.githubusercontent.com/24837412/101669586-c1804580-3a17-11eb-8f62-f4829200c5e6.png)
_without ssao_
![image](https://user-images.githubusercontent.com/24837412/101669522-a57ca400-3a17-11eb-9d47-d34aa53eefb4.png)
_with ssao_
![image](https://user-images.githubusercontent.com/24837412/101669538-af060c00-3a17-11eb-87ee-87d6095cd82d.png)
_ao buffer_

Feel free to leave suggestions and propose improvements!

**Comment by nem0** [comment]
I looks much better than current SSAO (no wonder, current SSAO is pretty simple and bad). In a lot of cases it's flawless. Only issue I had was SSAO on grass. Also it's perf was not great, but I understand it's not optimized yet.  If you manage to make it as fast as the old SSAO (or close enough), we can throw old SSAO away and keep just this. If it's not possible to fast, we can keep both and let __users__ pick per project.

**Comment by nem0** [comment]
There's AO channel in gbuffer now, I changed old SSAO to use it - AO does not influence direct lighting
https://github.com/nem0/LumixEngine/commit/63c24b1257053fd20f2632336f61b47f0c6b379c
